velox/functions/prestosql/StringFunctions.h

```diff
@@ -267,4 +267,85 @@ struct StrLPosFunction : public StrPosFunctionBase<T, true> {};
 template <typename T>
 struct StrRPosFunction : public StrPosFunctionBase<T, false> {};
 
+template <typename T>
+struct LevenshteinDistanceFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<int64_t>& result,
+      const arg_type<Varchar>& left,
+      const arg_type<Varchar>& right) {
+    std::vector<int32_t> leftCodePoints = stringImpl::stringToCodePoints(left);
+    std::vector<int32_t> rightCodePoints =
+        stringImpl::stringToCodePoints(right);
+    doCall<int32_t>(result, leftCodePoints, rightCodePoints);
+  }
+
+  FOLLY_ALWAYS_INLINE void callAscii(
+      out_type<int64_t>& result,
+      const arg_type<Varchar>& left,
+      const arg_type<Varchar>& right) {
+    std::vector<int8_t> leftCodePoints =
+        stringImpl::asciiStringToCodePoints(left);
+    std::vector<int8_t> rightCodePoints =
+        stringImpl::asciiStringToCodePoints(right);
+    doCall<int8_t>(result, leftCodePoints, rightCodePoints);
+  }
+
+  template <typename codePointT>
+  FOLLY_ALWAYS_INLINE void doCall(
+      out_type<int64_t>& result,
+      std::vector<codePointT> leftCodePoints,
```

Comment:
pass vectors as const & to avoid copying

here, pass these as raw pointers to avoid having to create std::vector in all-ascii cast